wbooks/src/app/screens/Library/index.tsx

```diff
@@ -1,25 +1,40 @@
 import React from 'react';
 import { FlatList, ListRenderItem } from 'react-native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import ScreenWrapper from '@components/ScreenWrapper';
 import SingleBook from '@components/SingleBook';
 import { BOOKS_MOCK } from '@constants/mockBooks';
+import SCREENS, { LibraryType } from '@constants/screens';
 import { Book } from '@interfaces/books';
+import { RootStackParamList } from '@interfaces/rootStackParamList';
 import { trimLineBreaks } from '@utils/stringUtils';
 
 import styles from './styles';
 
-function Library() {
+interface Props {
+  navigation: StackNavigationProp<RootStackParamList, LibraryType>;
+}
+
+function Library({ navigation }: Props) {
   const renderBooks: ListRenderItem<Book> = ({ item }) => (
-    <SingleBook bookCover={item.imageUrl} bookTitle={trimLineBreaks(item.title)} bookAuthor={item.author} />
+    <SingleBook
+      bookCover={item.imageUrl}
+      bookTitle={trimLineBreaks(item.title)}
+      bookAuthor={item.author}
+      onPressProp={() => navigation.navigate(SCREENS.BOOK_DETAILS)}
+    />
   );
   const keyExtractor = ({ id }: Book) => String(id);
 
   return (
-    <FlatList
-      style={styles.container}
-      data={BOOKS_MOCK}
-      renderItem={renderBooks}
-      keyExtractor={keyExtractor}
-    />
+    <ScreenWrapper>
```

Comment:
Same here, only the SafeArea. It's not necessary a wrapper or new component for this

Reply:
Refactored